pkg/logql/downstream_test.go

```diff
@@ -55,6 +55,14 @@ func TestMappingEquivalence(t *testing.T) {
 		{`avg_over_time({a=~".+"} | logfmt | unwrap value [1s])`, false},
 		{`avg_over_time({a=~".+"} | logfmt | unwrap value [1s]) by (a)`, true},
 		{`quantile_over_time(0.99, {a=~".+"} | logfmt | unwrap value [1s])`, true},
+		{
+			`
+			  (quantile_over_time(0.99,{a=~".+"} | logfmt | unwrap value[1s]) by (a) > 1)
+			and
+			  avg by (a)(rate({a=~".+"}[1s]))
+			`,
+			false,
+		},
```

Comment:
> I was expecting to see some new test cases here; do the existing ones already cover the unimplemented scenarios?

This would have caused the `unimplemented` error before.